### PR TITLE
Fix Scheduler MemorySlot mapping (issue #2704)

### LINKED ISSUE #2704: Scheduler needs to respect MemSlots in QuantumCircuit Measures
### Information

Known bug from my PR https://github.com/Qiskit/qiskit-terra/pull/2650

### What is the current behavior?
When passing a circuit through the basic scheduler (`qiskit.scheduler.schedule`), the resulting `Schedule` will always measure qubit results into the `MemorySlot` which matches the qubit index.
For example:
```
qc.measure(q[0], c[3])
```
will measure qubit 0 into `MemorySlot(0)` rather than `MemorySlot(3)` -- it will use the default Acquire from the cmd_def which is probably something like `Acquire([0, 1], [MemSlot(0), MemSlot(1)])`. The correct behaviour is to use the classical reg from the instruction and then modify the `AcquireInstruction` from the `Schedule` returned by the `cmd_def`. It's unclear what we should do with the other qubits that are measured within the same `meas_map` group, if the user doesn't specify those other qubits to be measured.

We need a follow up PR to handle this more carefully. The solution likely needs to update the helper function: `qiskit.scheduler.methods.basic.translate_gates_to_pulse_defs.get_measure_schedule`

### Suggested solutions
see above

### PR DESCRIPTION
**Pull request by lcapelluto**
### Summary

Closes #2704

### Details and comments

This is @SooluThomas's PR: 
https://github.com/Qiskit/qiskit-terra/pull/3415
with the tests cleaned up (rushed for Monday's release)

### CODE REVIEW

File: qiskit/scheduler/methods/basic.py [hunk 1]
```diff
@@ -140,31 +142,50 @@ def translate_gates_to_pulse_defs(circuit: QuantumCircuit,
     circ_pulse_defs = []
 
     cmd_def = schedule_config.cmd_def
-    measured_qubits = set()  # Collect qubits that would like to be measured
+    qubit_mem_slots = {}  # Map measured qubit index to classical bit index
 
     def get_measure_schedule() -> CircuitPulseDef:
```
Comment (taalexander) [inline qiskit/scheduler/methods/basic.py]: I think this should be made a standalone method but we can do this in a future PR. What do you think?

File: qiskit/scheduler/methods/basic.py [hunk 2]
```diff
@@ -140,31 +142,50 @@ def translate_gates_to_pulse_defs(circuit: QuantumCircuit,
...
         inst_qubits = [qubit.index for qubit in qubits]  # We want only the indices of the qubits
-        if any(q in measured_qubits for q in inst_qubits):
+        if any(q in qubit_mem_slots for q in inst_qubits):
             # If we are operating on a qubit that was scheduled to be measured, process that first
             circ_pulse_defs.append(get_measure_schedule())
         if isinstance(inst, Barrier):
             circ_pulse_defs.append(CircuitPulseDef(schedule=inst, qubits=inst_qubits))
         elif isinstance(inst, Measure):
-            measured_qubits.update(inst_qubits)
+            if (len(inst_qubits) != 1 and len(clbits) != 1):
```
Comment (taalexander) [inline qiskit/scheduler/methods/basic.py]: Won't this cause a failure if the number of qubits != classicalbit and > 1. Should this be `or`?

Comment (lcapelluto) [reply qiskit/scheduler/methods/basic.py]: yes, shoot, I didn't notice this section had changed. I'll update it

File: qiskit/scheduler/methods/basic.py (hunk 1 above)
Comment (lcapelluto) [reply qiskit/scheduler/methods/basic.py]: Ok, please make an issue:)

Comment (taalexander) [reply qiskit/scheduler/methods/basic.py]: Tracking in #3572.

**Pull request merged**